pkg/asset/agent/manifests/nmstateconfig_test.go

```diff
@@ -21,12 +25,49 @@ func TestNMStateConfig_Generate(t *testing.T) {
 		name           string
 		dependencies   []asset.Asset
 		expectedError  string
-		expectedConfig *aiv1beta1.NMStateConfig
+		expectedConfig []*aiv1beta1.NMStateConfig
 	}{
 		{
-			name:          "missing-config",
+			name: "missing-config",
```

Comment:
Another test case to include would be a valid AgentConfig, but without networkConfig specified.

Reply:
That causes a weird failure for that test 
```
expected: []*v1beta1.NMStateConfig{(*v1beta1.NMStateConfig)(0xc000774000)}
            	            	actual  : []*v1beta1.NMStateConfig{(*v1beta1.NMStateConfig)(0xc0007746e0)}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -44,4 +44,3 @@
            	            	    NetConfig: (v1beta1.NetConfig) {
            	            	-    Raw: (v1beta1.RawNetConfig) {
            	            	-    }
            	            	+    Raw: (v1beta1.RawNetConfig) <nil>
            	            	    }
```
I will add the test in a separate commit to this PR

Reply:
@andfasano :point_up:

Reply:
Discussed with @rwsu over the slack. When unmarshalled, the RawNetConfig gets set to a byte[] containing an empty interface causing this :point_up:  weird behavior. Removing this scenario.